export-ingester/export_ingester/sftp.py

```diff
@@ -19,11 +28,12 @@ async def pipe_exports(
         _mapper: Callable[[list[ExportModel]], bytes] | None = None,
     ):
         mapper = _mapper or partial(ExportList.dump_json, indent=4)
-        async with self.client.open(path, "wb") as f:
-            await f.write(mapper(exports))
+        async with self.conn.start_sftp_client() as sftp_client:
+            async with sftp_client.open(path, "wb") as f:
+                await f.write(mapper(exports))
 
 
-class SFTPClientFactory:
+class SSHClientFactory:
```

Comment:
Perhaps this class should have methods to return `SbatchClient` and `SFTPClient`?

Reply:
Well yes, I wasn't sure how to divide things. Because from what I see, I can do 'run' directly from the connection (so the way I understand it is that there already is a ssh client). But then from the connection I have to launch a sftp client. So in order to only have one connection started, I had decided to do the two things as part of a more general "SSH_client". Not sure if the vocabulary is right here but do you understand my problem?